src/region_growing.jl

```diff
@@ -0,0 +1,137 @@
+
+# An implementation of the Improved Seeded Region Growing Algorithm in Julia
+#
+# Citation:
+# Albert Mehnert, Paul Jackaway (1997), "An improved seeded region growing algorithm"
+# http://www.ee.bgu.ac.il/~itzik/IP5211/Other/Projects/P19_Seeded%20region%20growing.pdf
+
+function srg{CT<:Colorant}(img::AbstractArray{CT,2}, seeds::AbstractVector{Tuple{CartesianIndex{2}, Int}}, diff_fn::Function = (c1,c2)->(sum(abs,c1-c2)))
```

Comment:
There may need to be a bit of subtlety in the diff function due to unsignedness and arithmetic overflow:
```julia
julia> using FixedPointNumbers

julia> a, b = N0f8(0.1), N0f8(0.11)
(0.102N0f8, 0.11N0f8)

julia> a-b
0.996N0f8
```
That could potentially account for some oddities in your otherwise-impressive [demo](https://cloud.githubusercontent.com/assets/15063205/26763233/e4187b02-496c-11e7-828a-ace4f4230272.png).

You probably want to define an "accumulator type" that you cast pixels to before computing differences. There are some (old) examples in Images.jl; also note the existence of `FixedPointNumbers.floattype(N0f8)`.